src/main/java/net/clgd/ccemux/peripherals/PeripheralFactory.java

```diff
@@ -0,0 +1,14 @@
+package net.clgd.ccemux.peripherals;
+
+import dan200.computercraft.api.peripheral.IPeripheral;
+import net.clgd.ccemux.emulation.EmulatedComputer;
+
+import java.util.HashMap;
+import java.util.Map;
+
+@FunctionalInterface
+public interface PeripheralFactory<T extends IPeripheral> {
+	public static final Map<String, PeripheralFactory<?>> implementations = new HashMap<>();
```

Comment:
@Lignum a `register` method wouldn't change much, interface static methods can only be `public`.